src/pods/movies/components/movies-filter.component.tsx

```diff
@@ -1,4 +1,134 @@
 import * as React from 'react';
+import { TextField, MenuItem, createStyles, WithStyles, Button } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { MovieFilter, GenresList } from '../view-model';
+import { GenreEntity } from '../../../api/model';
+import { genresAPI } from '../../../api/genres-api'
 
-export const MoviesFilterComponent = () => 
-  <h3>Hello from Movies MoviesFilterComponent</h3>
\ No newline at end of file
+const styles = theme => createStyles({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    flexDirection: 'column',
+    marginRight: '30px',
+  },
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    marginBottom: theme.spacing.unit,
+    width: 300,
+  },
+  dense: {
+    marginTop: 19,
+  },
+  menu: {
+    width: 300,
+  },
+  button: {
+    margin: theme.spacing.unit,
+    width: 150,
+  },
+  input: {
+    display: 'none',
+  },
+});
+
+interface Props extends WithStyles<typeof styles> {
+  onApplyFilter: (movieFilter: MovieFilter) => void;
+}
+
+interface State {
+  movieFilter: MovieFilter;
+  genres: GenreEntity[];
+}
+
+export class MoviesFilterComponentInner extends React.Component<Props, State> {
+
+  state: State = {
+    movieFilter: {
+      title: '',
+      genre: '',
+      year: 2018,
+    },
+    genres: [],
+  };
+
+  componentDidMount() {
+    genresAPI.getAllGenres().then((genres : GenreEntity[]) => {
+      this.setState({
+        genres
+      });
+    })
+  }
+
+  updateTitle = (title: string) =>
+    this.setState({
+      movieFilter: {
+        ...this.state.movieFilter,
+        title
+      }
+    });
```

Comment:
We should go for a generic approach (in the future we could have more filters), not sure if this applies to combos

```typescript
  updateTitle = (fieldId: string) => (value: any) =>
    this.setState({
      movieFilter: {
        ...this.state.movieFilter,
        [fieldId]: value,
      }
    });
```